agent/connect/ca/provider_vault_auth_alicloud.go

```diff
@@ -0,0 +1,44 @@
+package ca
+
+import (
+	"fmt"
+
+	"github.com/aliyun/alibaba-cloud-sdk-go/sdk/auth/credentials/providers"
+	"github.com/hashicorp/consul/agent/structs"
+	"github.com/hashicorp/vault-plugin-auth-alicloud/tools"
+)
+
+func NewAliCloudAuthClient(authMethod *structs.VaultAuthMethod) (*VaultAuthClient, error) {
```

Comment:
There is no `legacyCheck` here like there is for the other auth methods. Will this break backwards compatibility for configurations using the `/login` API directly?

Reply:
The PMs don't think anyone is using Alicloud with mesh ATM and weren't even sure Alicloud was supported for Consul Mesh, but we had decided to go ahead and merge it as it seemed possible and the code was self contained.

I took the above to mean that backwards compatibility wasn't necessary for it, that we more wanted it there as a way forward. Does that make sense or do you think it'd be better to keep it in line with the others?

Reply:
Ok, thanks for the explanation. I feel like if it's not too much work to include the `legacyCheck` that we should do it, even if no one's using this and it's just for consistency across all of our auth method implementations. I'll leave it up to you :+1:

Reply:
I'll go ahead and add it... I had almost talked myself into it when writing about keeping it in line with the others. Seems like it should be the same (with legacyCheck) or it would need a comment why it didn't. I think just adding it will be easier. :)

Reply:
Added legacyCheck + a test for it.